File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include <regex>

using namespace std;

int solution(string s) {
    vector<string> numStr = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    
    regex r;
    for(int i = 0; i < numStr.size(); i++) {
        r = numStr[i];
        s = regex_replace(s, r, to_string(i));
    }
    return stoi(s);
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cs
using System;

public class Solution {
    public int solution(string s) {
        int answer = 0;
        string[] numStr = new string[10] {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

        for(int i = 0; i < numStr.Length; i++) {
            s = s.Replace(numStr[i], i.ToString());
        }
        
        answer = int.Parse(s);
        return answer;
    }
}